src/js/pages/ClientMirror.js

```diff
@@ -1,21 +1,50 @@
 import React from 'react';
+import { withRouter } from 'react-router-dom';
 import CustomCodeMirror from '../components/custom/CustomCodeMirror';
 import ActionBar from '../components/clientMirror/ActionBar';
 import ResultPane from '../components/clientMirror/ResultPane';
+const axios = require('axios');
 
 const initialState = require('../data/client-mirror-initial-state');
 const jsSdkVersions = require('../data/js-sdk-versions');
 
 class ClientMirror extends React.Component {
-  constructor() {
-    super();
+  constructor(props) {
+    super(props);
     this.state = {
```

Comment:
@marcioaffonso Here it says `constructor` comes under Mounting: "Mounting
These methods are called when an instance of a component is being created and inserted into the DOM"
Whilst `componentWIllReceiveProps` comes under Updating: "An update can be caused by changes to props or state. These methods are called when a component is being re-rendered:"
and below "React doesn't call componentWillReceiveProps with initial props during mounting"

So I think we need both. Source:
https://facebook.github.io/react/docs/react-component.html#the-component-lifecycle